modules/tensor_mechanics/src/auxkernels/GlobalDisplacementAux.C

```diff
@@ -37,19 +38,20 @@ GlobalDisplacementAux::GlobalDisplacementAux(const InputParameters & parameters)
   : AuxKernel(parameters),
     _scalar_global_strain(coupledScalarValue("scalar_global_strain")),
     _component(getParam<unsigned int>("component")),
-    _output_total_disp(getParam<bool>("output_total_displacement")),
-    _ndisp(getParam<bool>("output_total_displacement") ? coupledComponents("displacements") : 0),
-    _disp(_ndisp)
+    _output_global_disp(getParam<bool>("output_global_displacement")),
+    _dim(_mesh.dimension()),
+    _ndisp(coupledComponents("displacements")),
+    _disp(_ndisp),
+    _disp_var(_ndisp)
 {
-  if (_component > _mesh.dimension())
-    mooseError("The component ",
-               _component,
-               " does not exist for ",
-               _mesh.dimension(),
-               " dimensional problems");
+  if (_component > _dim)
+    mooseError("The component ", _component, " does not exist for ", _dim, " dimensional problems");
```

Comment:
Lets use `paramError` here for a better error message (that points to a line in the input file)